cli/cli_test.go

```diff
@@ -751,3 +744,111 @@ func TestGetEntitlementWithWrongTypeShowsError(t *testing.T) {
 	ctx := runner(newTstCtx(t, " "), "entitlement", "get", "actor", "swayze")
 	ctx.assertInfoErrContains("USAGE", "First parameter of 'get' must be user, group or app")
 }
+
+// - Oauth2 Application Templates
+
+// Helper to setup mock for the app template service
+func setupTemplateServiceMock() *mocks.OauthResource {
+	templServiceMock := new(mocks.OauthResource)
+	templateService = templServiceMock
+	return templServiceMock
+}
+
+func TestCanGetTemplate(t *testing.T) {
+	templServiceMock := setupTemplateServiceMock()
+	templServiceMock.On("Get", mock.Anything, "makesnow").Return()
+	testCliCommand(t, "template", "get", "makesnow")
+	templServiceMock.AssertExpectations(t)
+}
+
+// Helper to create template map
+func templateInfo(name, scope string, accessTokenTTL int) map[string]interface{} {
+	return map[string]interface{}{"scope": scope, "accessTokenTTL": accessTokenTTL,
+		"authGrantTypes": "authorization_code", "displayUserGrant": false,
+		"resourceUuid": "00000000-0000-0000-0000-000000000000", "tokenType": "Bearer",
+		"appProductId": name, "redirectUri": "horizonapi://oauth2",
+		"refreshTokenTTL": 2628000, "length": 32}
+}
+
+func TestCanAddTemplateWithDefaults(t *testing.T) {
+	templServiceMock := setupTemplateServiceMock()
+	templServiceMock.On("Add", mock.Anything, "olaf", templateInfo("olaf", "user profile email", 480)).Return()
+	testCliCommand(t, "template", "add", "olaf")
+	templServiceMock.AssertExpectations(t)
+}
+
+func TestCanAddTemplateWithOptions(t *testing.T) {
+	templServiceMock := setupTemplateServiceMock()
+	templServiceMock.On("Add", mock.Anything, "olaf", templateInfo("olaf", "snow", 0)).Return()
+	testCliCommand(t, "template", "add", "--scope", "snow", "--accessTokenTTL", "0", "olaf")
+	templServiceMock.AssertExpectations(t)
+}
+
+func TestCanDeleteTemplate(t *testing.T) {
```

Comment:
yes, will add. There are multiple other places that already test parameter counts, but adding this would be more complete for this command.